amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+
+    status |= vxQueryNode(node, VX_NODE_BORDER, &border, sizeof(border));
+    
+    border.mode = VX_BORDER_REPLICATE;
+
+    vxSetContextAttribute(context, VX_CONTEXT_IMMEDIATE_BORDER, &border, sizeof(border));
+
+    gaussian = vxCreatePyramid(context, levels + 1, VX_SCALE_PYRAMID_HALF, width, height, VX_DF_IMAGE_U8);
+    vxuGaussianPyramid(context, input, gaussian);
+
+    conv = vxCreateGaussian5x5Convolution(context);
+
+    level_width = width;
+    level_height = height;
+    gauss_cur = vxGetPyramidLevel(gaussian, 0);
+    gauss_next = vxGetPyramidLevel(gaussian, 1);
+    for (lev = 0; lev < levels; lev++)
+    {
+        pyr_gauss_curr_level_filtered = vxCreateImage(context, level_width, level_height, VX_DF_IMAGE_S16);
+        upsampleImage(context, level_width, level_height, gauss_next, conv, pyr_gauss_curr_level_filtered, &border);
+
+        pyr_laplacian_curr_level = vxGetPyramidLevel(laplacian, (vx_uint32)lev);
+        status |= vxuSubtract(context, gauss_cur, pyr_gauss_curr_level_filtered, policy, pyr_laplacian_curr_level);
+
+        if (lev == levels - 1)
+        {
+            vx_image tmp = vxGetPyramidLevel(gaussian, (vx_uint32) levels);
+            ownCopyImage(tmp, output);
+            vxReleaseImage(&tmp);
+            vxReleaseImage(&gauss_next);
+            vxReleaseImage(&gauss_cur);
+        }
+        else
+        {
+            /* compute dimensions for the next level */
+            level_width = (vx_uint32)ceilf(level_width * VX_SCALE_PYRAMID_HALF);
+            level_height = (vx_uint32)ceilf(level_height * VX_SCALE_PYRAMID_HALF);
+            /* prepare to the next iteration */
+            /* make the next level of gaussian pyramid the current level */
+            vxReleaseImage(&gauss_next);
+            vxReleaseImage(&gauss_cur);
+            gauss_cur = vxGetPyramidLevel(gaussian, (vx_uint32)lev + 1);
+            gauss_next = vxGetPyramidLevel(gaussian, (vx_uint32)lev + 2);
+
+        }
+
+        /* decrements the references */
+
+        status |= vxReleaseImage(&pyr_gauss_curr_level_filtered);
+        status |= vxReleaseImage(&pyr_laplacian_curr_level);
+    }
+
+    status |= vxReleasePyramid(&gaussian);
+    status |= vxReleaseConvolution(&conv);
+
+    return status;
+}
+
+#define VX_SCALE_PYRAMID_DOUBLE (2.0f)
+
+int HafCpu_LaplacianReconstruct_DATA_DATA_DATA
+    (
+        vx_node node, 
+        vx_pyramid laplacian,
+        vx_image input,
+        vx_image output
+    )
+{
+    vx_status status = VX_SUCCESS;
+
+    vx_context context = vxGetContext((vx_reference)node);
+
+    vx_size lev;
+    vx_size levels = 1;
+    vx_uint32 width = 0;
+    vx_uint32 height = 0;
+    vx_uint32 level_width = 0;
+    vx_uint32 level_height = 0;
+    vx_df_image format = VX_DF_IMAGE_S16;
+    vx_enum policy = VX_CONVERT_POLICY_SATURATE;
+    vx_border_t border;
+    vx_image filling = 0;
+    vx_image pyr_level = 0;
+    vx_image filter = 0;
+    vx_image out = 0;
```

Comment:
Codacy found an issue: [Variable 'out' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5686893)